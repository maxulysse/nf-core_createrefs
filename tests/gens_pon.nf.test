nextflow_pipeline {

    name "Test pipeline | gens"
    script "../main.nf"
    tag "pipeline"
    tag "pipeline_createpanelrefs"
    config "./gens_pon.config"

    test("Run gens test") {

        when {
            params {
                fasta                 = "https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/homo_sapiens/genome/genome.fasta"
                gens_bin_length       = 100
                gens_pon_name         = 'gens_pon'
                gens_readcount_format = "TSV"
                outdir                = "$outputDir"
                tools                 = 'gens'
            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            // stable_path: All files in ${params.outdir}/ with stable content
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // pipeline versions.yml file for multiqc from which Nextflow version is removed because we tests pipelines on multiple Nextflow versions
                    removeNextflowVersion("$outputDir/pipeline_info/nf_core_ceatepanelrefs_software_mqc_versions.yml"),
                    // All stable path name, with a relative path
                    stable_name,
                    // All files with stable contents
                    stable_path
                ).match() }
            )
        }
    }
}
