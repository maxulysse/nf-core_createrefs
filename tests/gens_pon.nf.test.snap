{
    "Run gens test": {
        "content": [
            7,
            {
                "GATK4_COLLECTREADCOUNTS": {
                    "gatk4": "4.5.0.0"
                },
                "GATK4_CREATEREADCOUNTPANELOFNORMALS": {
                    "gatk4": "4.5.0.0"
                },
                "GATK4_PREPROCESSINTERVALS": {
                    "gatk4": "4.5.0.0"
                },
                "PICARD_CREATESEQUENCEDICTIONARY": {
                    "picard": "3.3.0"
                },
                "SAMTOOLS_FAIDX": {
                    "samtools": 1.21
                },
                "Workflow": {
                    "nf-core/createpanelrefs": "v1.0dev"
                }
            },
            [
                "gens_pon",
                "gens_pon/createreadcountpanelofnormals",
                "gens_pon/createreadcountpanelofnormals/gens_pon.hdf5",
                "gens_pon/intervals",
                "gens_pon/intervals/genome.interval_list",
                "gens_pon/readcounts",
                "gens_pon/readcounts/sample1.tsv",
                "gens_pon/readcounts/sample2.tsv",
                "gens_pon/references",
                "gens_pon/references/genome.dict",
                "gens_pon/references/genome.fasta.fai",
                "multiqc",
                "multiqc/multiqc_data",
                "multiqc/multiqc_data/multiqc.log",
                "multiqc/multiqc_data/multiqc_citations.txt",
                "multiqc/multiqc_data/multiqc_data.json",
                "multiqc/multiqc_data/multiqc_software_versions.txt",
                "multiqc/multiqc_data/multiqc_sources.txt",
                "multiqc/multiqc_report.html",
                "pipeline_info",
                "pipeline_info/nf_core_ceatepanelrefs_software_mqc_versions.yml"
            ],
            [
                "sample1.tsv:md5,bfceab35109b04317f38f9cc1cde71ca",
                "sample2.tsv:md5,7141d08cdc26f6057557be9e23ef4365",
                "genome.fasta.fai:md5,3520cd30e1b100e55f578db9c855f685",
                "multiqc_citations.txt:md5,4c806e63a283ec1b7e78cdae3a923d4f"
            ]
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-04T19:29:15.733933882"
    }
}